src/main/java/de/hysky/skyblocker/skyblock/dwarven/CrystalsWaypoint.java

```diff
@@ -0,0 +1,128 @@
+import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderContext;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.command.argument.EnumArgumentType;
+import net.minecraft.entity.Entity;
+import net.minecraft.text.Text;
+import net.minecraft.text.TextCodecs;
+import net.minecraft.util.Formatting;
+import net.minecraft.util.StringIdentifiable;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.awt.*;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.function.ToDoubleFunction;
+
+public class CrystalsWaypoint extends Waypoint {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CrystalsWaypoint.class);
+
+    private static final Supplier<SkyblockerConfig.Waypoints> CONFIG = () -> SkyblockerConfigManager.get().general.waypoints;
+    static final Supplier<Type> TYPE_SUPPLIER = () -> CONFIG.get().waypointType;
+    final Category category;
+    final Text name;
+    private final Vec3d centerPos;
+
+    CrystalsWaypoint( Category category, Text name, BlockPos pos) {
+        super(pos, TYPE_SUPPLIER, category.colorComponents);
+        this.category = category;
+        this.name = name;
+        this.centerPos = pos.toCenterPos();
+    }
+
+    static ToDoubleFunction<CrystalsWaypoint> getSquaredDistanceToFunction(Entity entity) {
+        return crystalsWaypoint -> entity.squaredDistanceTo(crystalsWaypoint.centerPos);
+    }
+
+    static Predicate<CrystalsWaypoint> getRangePredicate(Entity entity) {
+        return crystalsWaypoint -> entity.squaredDistanceTo(crystalsWaypoint.centerPos) <= 36D;
+    }
+
+    @Override
+    public boolean shouldRender() {
+        return super.shouldRender() ;
+    }
+
+
+    @Override
+    public boolean equals(Object obj) {
+        return super.equals(obj) || obj instanceof CrystalsWaypoint other && category == other.category && name.equals(other.name) && pos.equals(other.pos);
+    }
+
+    /**
+     * Renders the secret waypoint, including a waypoint through {@link Waypoint#render(WorldRenderContext)}, the name, and the distance from the player.
+     */
+    @Override
+    public void render(WorldRenderContext context) {
+        super.render(context);
+
+
+        Vec3d posUp = centerPos.add(0, 1, 0);
+        RenderHelper.renderText(context, name, posUp, true);
+        double distance = context.camera().getPos().distanceTo(centerPos);
+        RenderHelper.renderText(context, Text.literal(Math.round(distance) + "m").formatted(Formatting.YELLOW), posUp, 1, MinecraftClient.getInstance().textRenderer.fontHeight + 1, true);
+
+    }
+
+
+    /**
+     * enum for the different waypoints used int the crystals hud each with a {@link Category#name} and associated {@link Category#color}
+     */
+    enum Category implements StringIdentifiable {
```

Comment:
Separate the different words in the enum constants with an underscore `_`

Eg: `JUNGLE_TEMPLE`, `LOST_PRECURSOR_CITY`